photon-client/src/main/scala/com/linkedin/photon/ml/estimators/GameParams.scala

```diff
@@ -494,6 +503,16 @@ object GameParams {
           s"default: ${defaultParams.checkData}")
         .foreach(x => params.checkData = x)
 
+      opt[String](INPUT_COLUMNS_NAMES)
+        .text(s"A map of the input columns names, in format: response:label|offset:offset|uid:uid ... ")
+        .foreach(x =>
+          x.split("\\|")
+            .foreach { line =>
+              val Array(key, value) = line.split(":").map(_.trim)
```

Comment:
This will generate a confusing `MatchError` in the case of a malformed entry (e.g. `"response:label:hello"`). It would be better to match on this and generate a more helpful error in the case that it's wrong, e.g.:

```scala
  .foreach { line =>
    line.split(":").map(_.trim) match {
      case Array(key, value) => params.inputColumnsNames.updated(InputColumnsNames.withName(key), value)
      case other => s"Whoops, couldn't interpret '$other'"
    }
  }
```